source/extensions/quic_listeners/quiche/active_quic_listener.cc

```diff
@@ -136,6 +164,43 @@ void ActiveQuicListener::shutdownListener() {
   quic_dispatcher_->StopAcceptingNewConnections();
 }
 
+absl::optional<uint32_t> ActiveQuicListener::destination(const Network::UdpRecvData& data,
```

Comment:
I agree that performance won't be very good.

In the case where in-kernel routing is possible, that is always done, and no re-routing will occur.

But it's useful to have a software fallback, so the code is at least functional in the case of an old kernel or non-linux platform. For this first pass, I just wanted to make it functional, even though performance will be inadequate for many use cases. I'm in favor of trying to make it faster in future PRs.

But for the primary use-case of running on a recent linux kernel, this code no-ops immediately, so it isn't a regression on current behavior.

Reply:
If so, can you comment that this code is not performant enough to run in production and file a tracking issue for it?

Reply:
Can we do more than comment? Can there be a warning at startup if we are using this?

Edit: nevermind I see we have the warning log.

Reply:
Can you also add a warning here to specifically state that it is using this UDP packet routine impl that is problematic rather than not using BPF, because there might be other impl which can do efficient user space packet routine.

Reply:
@danzh2010 can you clarify what you're asking for? A bigger warning than what I already added? https://github.com/envoyproxy/envoy/pull/13086/commits/decfdd0e7f37d576f1992c815b812a342c610756

Reply:
That commit only add a comment, can you also add a warning or even error log for using this impl in production?

Reply:
There is a warning-level message printed for when this impl is in use, here: https://github.com/envoyproxy/envoy/pull/13086/files#diff-db91269a6c7fbac4e9fab713910c85eeR283

Reply:
ACK, I missed the change of wording there. Thx!